charts/metering-ansible-operator/templates/crds/meteringconfig.crd.yaml

```diff
@@ -1549,6 +1549,47 @@ spec:
                                   properties:
                                     useInstanceCredentials:
                                       type: boolean
+                            prometheus:
+                              type: object
+                              oneOf:
+                              - required:
+                                - enabled
+                                - uri
+                                properties:
+                                  enabled:
+                                    enum:
+                                    - true
+                              - properties:
+                                  enabled:
+                                    enum:
+                                    - false
+                                allOf:
+                                - not:
+                                    required:
+                                      - uri
+                              properties:
+                                enabled:
+                                  description: Whether or not to enable the Presto-Prometheus connector.
+                                  type: boolean
+                                uri:
+                                  description: URI for Prometheus.
+                                  type: string
+                                  format: uri
+                                chunkSizeDuration:
+                                  description: Default size of each Prometheus query chunk.
+                                  type: string
+                                maxQueryRangeDuration:
+                                  description: Default range for Prometheus query, divided into chunkSizeDuration chunks.
+                                  type: string
+                                cacheDuration:
+                                  description: Time to live for Presto checking connector properties values.
+                                  type: string
+                                bearer:
```

Comment:
Copying this discussion over so it can be addressed in this PR:
timflannagan1 4 days ago Member
@chancez do you think this should just bearerTokenFile for the time being, or do you think there will be more authentication/bearer configurations in the future that warrants it's own section?

 @chancez
chancez 4 days ago Member
Good question. I guess generally speaking, being able to specify a path directly isn't very useful actually since it'll be coming either from the service account secret, or an arbitrary user supplied secret.

In the case of a user supplied secret, they wouldn't likely have much choice in the location of the file, and we would just mount it to a known location.

So perhaps we should have an. A useServiceAccountToken bool option and eventually (not now) we could add a secretName config for supplying your own credentials.

WDYT?

 @chancez
chancez 4 days ago Member
Good question. I guess generally speaking, being able to specify a path directly isn't very useful actually since it'll be coming either from the service account secret, or an arbitrary user supplied secret.

In the case of a user supplied secret, they wouldn't likely have much choice in the location of the file, and we would just mount it to a known location.

So perhaps we should have an. A useServiceAccountToken bool option and eventually we could add a secretName config for supplying your own credentials.

Reply:
Made the chart changes for this in: c64a8ec1
Thinking it's likely we don't want to check `useServiceAccountToken` on the ansible validation side until we provide ability to specify an alternative token location.
Should we have a story for that so we don't forget to do it?

Reply:
Yeah let's hold off on creating a story for now.